doc/source/whatsnew/v0.24.0.rst

```diff
@@ -1309,6 +1309,7 @@ Deprecations
 - :meth:`Series.clip_lower`, :meth:`Series.clip_upper`, :meth:`DataFrame.clip_lower` and :meth:`DataFrame.clip_upper` are deprecated and will be removed in a future version. Use ``Series.clip(lower=threshold)``, ``Series.clip(upper=threshold)`` and the equivalent ``DataFrame`` methods (:issue:`24203`)
 - :meth:`Series.nonzero` is deprecated and will be removed in a future version (:issue:`18262`)
 - Passing an integer to :meth:`Series.fillna` and :meth:`DataFrame.fillna` with ``timedelta64[ns]`` dtypes is deprecated, will raise ``TypeError`` in a future version.  Use ``obj.fillna(pd.Timedelta(...))`` instead (:issue:`24694`)
+- :attr:``CategoricalAccessor.categorical``, :attr:``CategoricalAccessor.name`` and :attr:``CategoricalAccessor.index`` have been deprecated and replaced by private :attr:``CategoricalAccessor._parent``, :attr:``CategoricalAccessor._name`` and :attr:``CategoricalAccessor._index``, respectively. (:issue:`24751`).
```

Comment:
I don't think any of these links will work, as they aren't part of the API docs. We also don't want to direct people to use private methods. Rather, I'd say

    ``Series.cat.categorical``, ``Series.cat.name`` and ``Sersies.cat.index`` have been deprecated. Use
    the attributes on ``Series.cat`` or ``Series`` directly.